src/watchdog/observers/inotify_c.py

```diff
@@ -406,7 +410,7 @@ def _add_dir_watch(self, path: bytes, mask: int, *, recursive: bool) -> None:
             for root, dirnames, _ in os.walk(path):
                 for dirname in dirnames:
                     full_path = os.path.join(root, dirname)
-                    if os.path.islink(full_path):
+                    if os.path.islink(full_path) and not self._follow_symlink:
```

Comment:
For performances reasons, I would swap those checks since their evaluation is lazy, and checking the instance attribute will likely be faster & less resources consuming. WDYT?